src/main/java/com/amalitech/gpuconfigurator/config/ApplicationConfig.java

```diff
@@ -0,0 +1,69 @@
+package com.amalitech.gpuconfigurator.config;
+
+import com.amalitech.gpuconfigurator.repository.UserRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.AuthenticationProvider;
+import org.springframework.security.authentication.dao.DaoAuthenticationProvider;
+import org.springframework.security.config.annotation.authentication.configuration.AuthenticationConfiguration;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+
+@Configuration
+@RequiredArgsConstructor
+public class ApplicationConfig {
+
+
+    private final UserRepository repository;
+
+    /**
+     * Creates a UserDetailsService bean that retrieves user details from the UserRepository.
+     *
+     * @return An implementation of UserDetailsService.
+     */
+    @Bean
+    public UserDetailsService userDetailsService(){
```

Comment:
You can have this UserDetailsService bean in your UserService, and use that in the Authentication provider, this will help you take out the UserRepository dependency from this configuration. It'll be rather nice to interact with a service from the configuration that couple the configuration to a repository.

Reply:
The refactor is causing a lot of breaking changes. I'll like to keep it in ApplicationConfig for now. So that the others can make use of my service.